base/base/defines.h

```diff
@@ -28,7 +28,7 @@
 #define NO_INLINE __attribute__((__noinline__))
 #define MAY_ALIAS __attribute__((__may_alias__))
 
-#if !defined(__x86_64__) && !defined(__aarch64__) && !defined(__PPC__) && !(defined(__riscv) && (__riscv_xlen == 64))
+#if !defined(__x86_64__) && !defined(__s390x__) && !defined(_S390X__) && !defined(__aarch64__) && !defined(__PPC__) && !(defined(__riscv) && (__riscv_xlen == 64))
```

Comment:
Checking twice for S390X is weird. Let's use whatever *single* definition compilers set on s390x.

(same comment also applies to similar places in this PR)